reliability/retry/retry.go

```diff
@@ -9,35 +9,30 @@ import (
 // Action function to execute in retry.
 type Action func() (interface{}, error)
 
-// Retry pattern with retries and optional delay.
+// Retry pattern with attempts and optional delay.
 type Retry struct {
-	retries int
-	delay   time.Duration
+	attempts int
+	delay    time.Duration
 }
 
 // New constructor.
-func New(retries int, delay time.Duration) (*Retry, error) {
-	if retries < 0 {
-		return nil, errors.New("retries should be zero or positive")
+func New(attempts int, delay time.Duration) (*Retry, error) {
+	if attempts <= 1 {
+		return nil, errors.New("attempts should be greater than 1")
```

Comment:
or equal?

Reply:
I have defined this validation so that it does not accept less than two attempts based on the fact that the retrial only makes sense when we have at least two attempts. 

My thinking is that supporting one total attempt as a valid input means that we won't be retrying anything. So why use the retrial pattern in the first place?

Do you think we have any use cases that will require accepting one attempt?

Reply:
Yes makes sense!